c/tree.c

```diff
@@ -633,34 +645,147 @@ LOCAL void alias_ipv4_networks(MMDBW_tree_s *tree)
+    //
+    // TODO(wstorey@maxmind.com): 64:ff9b:1::/48 should be in
+    // alias_ipv6_to_ipv4?
+
+    { .ipstr = "100::",      .prefix_length = 64 },
+
+    // 2001::/23 is reserved. We include all of it here other than 2001::/32
+    // as it is Teredo which is globally routable.
+    { .ipstr = "2001:1::",   .prefix_length = 32 },
+    { .ipstr = "2001:2::",   .prefix_length = 31 },
+    { .ipstr = "2001:4::",   .prefix_length = 30 },
+    { .ipstr = "2001:8::",   .prefix_length = 29 },
+    { .ipstr = "2001:10::",  .prefix_length = 28 },
+    { .ipstr = "2001:20::",  .prefix_length = 27 },
+    { .ipstr = "2001:40::",  .prefix_length = 26 },
+    { .ipstr = "2001:80::",  .prefix_length = 25 },
+    { .ipstr = "2001:100::", .prefix_length = 24 },
+
+    { .ipstr = "2001:db8::", .prefix_length = 32 },
+    // 2002::/16 - 6to4, part of alias_ipv6_to_ipv4
+    // 2620:4f:8000::/48 is routable I believe
+    { .ipstr = "fc00::",     .prefix_length = 7  },
+    { .ipstr = "fe80::",     .prefix_length = 10 },
+    // Multicast
+    { .ipstr = "ff00::",     .prefix_length = 8  },
+};
+
+// Insert a FIXED_EMPTY record for all private and reserved networks.
+//
+// This is to avoid the case where we might otherwise accidentally add
+// information about such networks.
+LOCAL void insert_reserved_networks_as_fixed_empty(MMDBW_tree_s *tree)
+{
+    insert_networks_as_fixed_empty(tree, reserved_networks_ipv4,
+                                   sizeof(reserved_networks_ipv4) /
+                                   sizeof(struct network));
+
+    if (tree->ip_version == 6) {
+        insert_networks_as_fixed_empty(tree, reserved_networks_ipv6,
+                                       sizeof(reserved_networks_ipv6) /
+                                       sizeof(struct network));
+    }
+}
+
+// Insert a FIXED_EMPTY record for each network.
+LOCAL void insert_networks_as_fixed_empty(MMDBW_tree_s *tree,
+                                          struct network const *const networks,
+                                          const size_t num_networks)
+{
+    for (size_t i = 0; i < num_networks; i++) {
+        MMDBW_network_s resolved_network = resolve_network(tree,
+                                                           networks[i].ipstr,
+                                                           networks[i].prefix_length);
+
+        MMDBW_record_s record = {
+            .type = MMDBW_RECORD_TYPE_FIXED_EMPTY,
+        };
+
+        MMDBW_status const status = insert_record_for_network(tree,
+                                                              &resolved_network,
+                                                              &record,
+                                                              MMDBW_MERGE_STRATEGY_NONE,
+                                                              true);
+
+        free_network(&resolved_network);
+
+        if (MMDBW_SUCCESS != status) {
+            croak(
```

Comment:
We would leak memory if this happened. However, it seems like it would only happen on a programming error in the writer code.

See comment on `croak` below.

Reply:
Yeah, this one we should be pretty safe on not happening